docs/lab/00.md

```diff
@@ -0,0 +1,516 @@
+
+To define a structure, we enter the `struct` keyword and name the entire structure. The name of a structure should describe the meaning of the data elements grouped together. Then, within curly brackets, we define the names and types of the data, which we call **fields**.
+
+```rust
+struct User {
+    active: bool,
+    username: String,
+    email: String,
+    sign_in_count: u64,
+}
+```
+
+To use a structure after having defined it, we create an **instance** of this structure by specifying concrete values ​​for each of the fields. We create an instance by specifying the structure **name , then add curly braces containing key:value pairs** , where the keys are the field names and the values ​​are the data we want to store in those fields.
+
+```rust
+fn main() {
+    let user1 = User {
+        active: true,
+        username: String::from("someusername123"),
+        email: String::from("someone@example.com"),
+        sign_in_count: 1,
+    };
+}
+```
+
+To access a certain member of the struct we use this syntax:
+
+```rust
+fn main() {
+    let mut user1 = User {
+        active: true,
+        username: String::from("someusername123"),
+        email: String::from("someone@example.com"),
+        sign_in_count: 1,
+    };
+ 
+    user1.email = String::from("anotheremail@example.com")
+}
+```
+
+:::tip 
+
+Note that the entire instance must be **editable**  ; Rust **doesn't allow us** to mark only certain fields as mutable!
+
+:::
+
+As with any expression, we can construct a new instance of the structure as the last expression in the function body to implicitly return this new instance.
+
+```rust
+fn build_user(email: String, username: String) -> User {
+    User {
+        active: true,
+        username: username,
+        email: email,
+        sign_in_count: 1,
+    }
+}
+```
+#### Tuple structs
+
+Rust also supports structures that resemble tuples, called **tuple structs** . Tuple structures have the additional meaning provided by the structure name but do not have names associated with their fields; instead, they just have the field types. Tuple structures are useful when you want to name the entire tuple and make it a different type from other tuples, and when naming each field as in a regular structure would be wordy or redundant.
+
+```rust
+struct Color(i32, i32, i32);
+struct Point(i32, i32, i32);
+ 
+fn main() {
+    let black = Color(0, 0, 0);
+    let origin = Point(0, 0, 0);
+}
+```
+
+
+:::tip 
+
+For a better understanding, please read [chapter 5](https://doc.rust-lang.org/book/ch05-00-structs.html) of the documentation.
+
+:::
+
+### String
+Rust has only one type of string in the base language, which is the string slice str which is usually seen in its borrowed form **&str** .
+
+The **String** type , which is provided by the Rust standard library rather than encoded in the main language, is a scalable, mutable, and owned UTF-8 encoded string type .
```

Comment:
```suggestion
The `String` type , which is provided by the Rust standard library rather than encoded in the main language, is a scalable, mutable, and owned UTF-8 encoded string type .
```